### ISSUE TITLE
Advanced indexing (numpy) not supported

### ISSUE THREAD

**Issue by Strateus** [pfnet/chainer#711]
Any plans on this?
```
  File "train_test.py", line 177, in train_epoch
    loss.backward()
  File "/root/anaconda2/lib/python2.7/site-packages/chainer-1.5.0.2-py2.7-linux-x86_64.egg/chainer/variable.py", line 277, in backward
    gxs = func.backward(in_data, out_grad)
  File "/root/Code/TripleChain/loss.py", line 111, in backward
    dLda[self.Li==0.] = 0.
  File "cupy/core/core.pyx", line 1037, in cupy.core.core.ndarray.__setitem__ (cupy/core/core.cpp:20894)
  File "cupy/core/core.pyx", line 969, in cupy.core.core.ndarray.__getitem__ (cupy/core/core.cpp:19883)
ValueError: Advanced indexing is not supported
```

Or maybe how i can overcome this temporary?

**Comment by beam2d** [comment]
We have no concrete plan on advanced indexing yet, though we want to support it in the future. It seems you want to fill 0 by boolean indexing. You can use [copyto](http://docs.chainer.org/en/stable/cupy-reference/manipulation.html#cupy.copyto) to do that. I found that the current implementation does not accept Python scalars as src, so pass a numpy scalar explicitly (I've raised [another issue](https://github.com/pfnet/chainer/issues/720)):
```
xp.copyto(dLda, dLda.dtype.type(0), where=self.Li==0.)
```

**Comment by Strateus** [comment]
great, thank you!

**Comment by yuyu2172** [comment]
Advanced indexing is implemented in #2203.
